ProcessLib/BoundaryCondition/UniformDirichletBoundaryCondition.cpp

```diff
@@ -11,70 +11,57 @@
 
 #include <algorithm>
 #include <vector>
-
 #include <logog/include/logog.hpp>
 
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "BoundaryConditionConfig.h"
+
 namespace ProcessLib
 {
-UniformDirichletBoundaryCondition::UniformDirichletBoundaryCondition(
-    GeoLib::GeoObject const& geometry, BaseLib::ConfigTree const& config)
-    : _geometry(geometry)
+std::unique_ptr<UniformDirichletBoundaryCondition>
+createUniformDirichletBoundaryCondition(
+    BoundaryConditionConfig const& config,
+    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+    NumLib::LocalToGlobalIndexMap const& dof_table,
+    int const variable_id)
 {
     DBUG("Constructing UniformDirichletBoundaryCondition from config.");
     //! \ogs_file_param{boundary_condition__type}
-    config.checkConfigParameter("type", "UniformDirichlet");
+    config.config.checkConfigParameter("type", "UniformDirichlet");
 
     //! \ogs_file_param{boundary_condition__UniformDirichlet__value}
-    _value = config.getConfigParameter<double>("value");
-    DBUG("Using value %g", _value);
-}
+    auto const value = config.config.getConfigParameter<double>("value");
+    DBUG("Using value %g", value);
 
-UniformDirichletBoundaryCondition::UniformDirichletBoundaryCondition(
-    GeoLib::GeoObject const& geometry, double value)
-    : _value(value), _geometry(geometry)
-{
-    DBUG("Constructed UniformDirichletBoundaryCondition using value %g",
-         _value);
-}
-
-/// Initialize Dirichlet type boundary conditions.
-/// Fills in global_ids of the particular geometry of the boundary condition
-/// and the corresponding values.
-/// The ids and the constant values are then used to construct DirichletBc
-/// object.
-void UniformDirichletBoundaryCondition::initialize(
-    MeshGeoToolsLib::MeshNodeSearcher& searcher,
-    NumLib::LocalToGlobalIndexMap const& dof_table,
-    int const component_id,
-    int const variable_id,
-    DirichletBc<GlobalIndexType>& bc)
-{
     // Find nodes' ids on the given mesh on which this boundary condition
     // is defined.
-    std::vector<std::size_t> ids = searcher.getMeshNodeIDs(_geometry);
+    std::vector<std::size_t> ids = searcher.getMeshNodeIDs(config.geometry);
+
+    NumLib::IndexValueVector<GlobalIndexType> bc;
 
     // convert mesh node ids to global index for the given component
     bc.ids.reserve(bc.ids.size() + ids.size());
     bc.values.reserve(bc.values.size() + ids.size());
-    for (auto& id : ids)
-    {
+    for (auto& id : ids) {
         MeshLib::Location l(searcher.getMeshId(), MeshLib::MeshItemType::Node,
                             id);
         // TODO: that might be slow, but only done once
         const auto g_idx =
-            dof_table.getGlobalIndex(l, variable_id, component_id);
+            dof_table.getGlobalIndex(l, variable_id, config.component_id);
```

Comment:
Looking at this issue from the create methods side: var_id comes from the process, which only knows about the order of its ProcessVariables, but not, which component gets which BC; component_id on the other hand comes directly from the BC config.